blog/content/post/2024-12-13-ram-data-decay-research.md

```diff
@@ -0,0 +1,357 @@
+caption="19.4 &#8451;, 30s power off time, 23.39% changed bits" >}}
+
+{{< figure src="/img/ram_decay_msi_3501.png"
+caption="19.3 &#8451;, 40s power off time, 35.01% changed bits" >}}
+
+{{< figure src="/img/ram_decay_msi_4281.png"
+caption="19.4 &#8451;, 50s power off time, 42.81% changed bits" >}}
+
+{{< figure src="/img/ram_decay_msi_4622.png"
+caption="19.3 &#8451;, 60s power off time, 46.22% changed bits" >}}
+
+{{< figure src="/img/ram_decay_msi_4771.png"
+caption="19.3 &#8451;, 70s power off time, 47.71% changed bits" >}}
+
+{{< figure src="/img/ram_decay_msi_4857.png"
+caption="19.2 &#8451;, 80s power off time, 48.57% changed bits" >}}
+
+{{< figure src="/img/ram_decay_msi_4896.png"
+caption="19.2 &#8451;, 90s power off time, 48.96% changed bits" >}}
+
+{{< figure src="/img/ram_decay_msi_4926.png"
+caption="19.2 &#8451;, 100s power off time, 49.26% changed bits" >}}
+
+{{< figure src="/img/ram_decay_msi_4943.png"
+caption="19.1 &#8451;, 110s power off time, 49.43% changed bits" >}}
+
+{{< figure src="/img/ram_decay_msi_4951.png"
+caption="19.2 &#8451;, 120s power off time, 49.51% changed bits" >}}
+
+With these measurement points, it was possible to construct a chart of preserved
+bits depending on power off time:
+
+{{< figure src="/img/ram_decay_msi_summary.png" >}}
+
+## Conclusions
+
+Both platforms show that the data decays with time, but the rate is
+significantly different between tested platforms. What took less than a second
+on DDR5 laptop, wasn't achieved even after two minutes on DDR4 desktop.
+
+On both platforms, impact of DIMM topology can be seen. In case of DDR5 SODIMM,
+distribution of changes in bits 0-31 roughly aligns with bits 32-63, especially
+with average values. Lower half has more 0 to 1 transitions, which in higher
+half is mirrored by 1 to 0 transitions. Border between DRAMs is also visible,
+although it isn't that clear, contrary to DDR4 DIMM. Results from the latter
+show a clear difference between groups of 8 consecutive bits, with average
+almost constant within the group.
+
+During a given run, some DRAMs start loosing their data faster than others. It
+looks like this isn't necessarily correlated with their maximal preservation
+time - take a look at group of bits 40-47 in DDR4. For delays in range 10-50
+seconds, it is always the one that looses the biggest part of its content, but
+after that, it doesn't get worse as fast as other groups do.
+
+For DDR4, chart of percentage of lost bits in function of time shows that the
+amount of lost data rises quickly with lower delays, but then starts to slow
+down when it gets close to 50%. The plot is also flattened on the left side,
+which may suggest that some constant, systematic error is added to time
+measurements. In fact, after applying back the power, it takes about 5 seconds
+before power LED gets lit on that platform.
+
+## Summary
+
+We've learned that RAM needs time to lost its content, and that this time varies
```

Comment:
```suggestion
We've learned that RAM needs time to lose its content, and that this time varies
```

Reply:
https://github.com/3mdeb/news-and-ideas/pull/621/commits/478ab8793dac42e71f71d1b2f30aeb7b92d053c6